src/software/geom/util_test.cpp

```diff
@@ -1529,3 +1349,210 @@ TEST(GeomUtilTest, test_find_closest_point_many_points)
+
+
+    // Circle 2
+    EXPECT_NEAR(proj_segments.back().getSegStart().x(), 1.92, 0.01);
+    EXPECT_DOUBLE_EQ(proj_segments.back().getSegStart().y(), 5.0);
+    EXPECT_NEAR(proj_segments.back().getEnd().x(), 0.62, 0.01);
+    EXPECT_DOUBLE_EQ(proj_segments.back().getEnd().y(), 5.0);
+}
+
+TEST(GeomUtilTest, test_project_circles_multiple_circles_one_has_zero_projection)
+{
+    Point reference = Point(0, 0);
+    Circle circle1  = Circle(Point(-1, 4), 0.5);
+    Circle circle2  = Circle(Point(1, 4), 0.5);
+    Circle circle3  = Circle(Point(5, -5), 1);
+    Segment segment = Segment(Point(5, 5), Point(-5, 5));
+
+    std::vector<Segment> proj_segments =
+        projectCirclesOntoSegment(segment, {circle1, circle2, circle3}, reference);
+
+    EXPECT_EQ(proj_segments.size(), 2);
+    EXPECT_NEAR(proj_segments.front().length(), 1.30, 0.01);
+
+    // Circle 1
+    EXPECT_NEAR(proj_segments.front().getSegStart().x(), -0.62, 0.01);
+    EXPECT_DOUBLE_EQ(proj_segments.front().getSegStart().y(), 5.0);
+    EXPECT_NEAR(proj_segments.front().getEnd().x(), -1.92, 0.01);
+    EXPECT_DOUBLE_EQ(proj_segments.front().getEnd().y(), 5.0);
+
+
+    // Circle 2
+    EXPECT_NEAR(proj_segments.back().getSegStart().x(), 1.92, 0.01);
+    EXPECT_DOUBLE_EQ(proj_segments.back().getSegStart().y(), 5.0);
+    EXPECT_NEAR(proj_segments.back().getEnd().x(), 0.62, 0.01);
+    EXPECT_DOUBLE_EQ(proj_segments.back().getEnd().y(), 5.0);
+}
+
+TEST(GeomUtilTest, test_get_empty_space_between_segment_no_space)
+{
+    Segment reference = Segment(Point(-10, 0), Point(10, 0));
+
+    Segment seg1 = Segment(Point(-10, 0), Point(0, 0));
+    Segment seg2 = Segment(Point(0, 0), Point(10, 0));
+
+    std::vector<Segment> segs = {seg1, seg2};
+
+    std::vector<Segment> open_segs = getEmptySpaceWithinParentSegment(segs, reference);
+
+    EXPECT_EQ(open_segs.size(), 0);
+}
+TEST(GeomUtilTest, test_get_empty_space_between_segment_2_blocks)
+{
+    Segment reference = Segment(Point(-10, 0), Point(10, 0));
+
+    Segment seg1 = Segment(Point(-8, 0), Point(0, 0));
+    Segment seg2 = Segment(Point(0, 0), Point(8, 0));
+
+    std::vector<Segment> segs = {seg1, seg2};
+
+    std::vector<Segment> open_segs = getEmptySpaceWithinParentSegment(segs, reference);
+
+    EXPECT_EQ(open_segs.size(), 2);
+
+    EXPECT_EQ(open_segs[0].length(), 2);
+    EXPECT_EQ(open_segs[1].length(), 2);
+}
+TEST(GeomUtilTest, test_get_empty_space_between_segment_2_blocks_3_open)
+{
+    Segment reference = Segment(Point(-10, 0), Point(10, 0));
+
+    Segment seg1 = Segment(Point(-8, 0), Point(-2, 0));
+    Segment seg2 = Segment(Point(2, 0), Point(8, 0));
+
+    std::vector<Segment> segs = {seg1, seg2};
+
+    std::vector<Segment> open_segs = getEmptySpaceWithinParentSegment(segs, reference);
+
+    EXPECT_EQ(open_segs.size(), 3);
+
+    EXPECT_EQ(open_segs[0].length(), 2);
+    EXPECT_EQ(open_segs[1].length(), 4);
+    EXPECT_EQ(open_segs[2].length(), 2);
+}
```

Comment:
the missing newline is back (ie. it's missing again)

Reply:
Pesky NLs